extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -67,6 +71,37 @@ export class AmpSelector extends AMP.BaseElement {
     }
   }
 
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    mutations.forEach(mutation => {
+      const newValue = mutation.newValue;
+      switch (mutation.name) {
+        case 'selected':
+          this.clearAllSelections_();
+          if (newValue) {
+            // Create a query with an attribute selector for every
+            // comma-delimited option value in `newValue`.
+            const options = String(mutation.newValue).split(',');
+            const selectors = [];
+            for (let i = 0; i < options.length; i++) {
+              // Only use first value if multiple selection is disabled.
+              if (i > 0 && !this.isMultiple_) {
+                break;
+              }
+              selectors.push(`[option='${options[i]}']`);
+            }
+            const query = selectors.join(',');
+            const elements = this.element.querySelectorAll(query);
```

Comment:
on a second thought 
lets loop through `this.options_` and see if the value is in the `newSelectionList` and `setSelection` or `clearSelection`